geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/Frequency.scala

```diff
@@ -0,0 +1,212 @@
+  *                  (note: first 2 bits do not hold any info)
+  *                  for date types, this is the number of milliseconds to group for binning
+  *                  for number types, this is the number of digits that will be grouped together
+  *                  for floating point types, this is the number of decimal places that will be considered
+  *                  for string types, this is the number of characters that will be considered
+  * @param ct class tag
+  * @tparam T type parameter, should match the type binding of the attribute
+  */
+class Frequency[T](val attribute: Int,
+                   val precision: Int,
+                   val eps: Double = 0.005,
+                   val confidence: Double = 0.95)(implicit ct: ClassTag[T]) extends Stat {
+
+  override type S = Frequency[T]
+
+  private [stats] val sketch: CountMinSketch = new CountMinSketch(eps, confidence, Frequency.Seed)
+
+  private val addAttribute = Frequency.add[T](ct.runtimeClass.asInstanceOf[Class[T]], precision)
+  private val getCount = Frequency.count[T](ct.runtimeClass.asInstanceOf[Class[T]], precision)
+
+  /**
+    * Gets the count for a given value
+    *
+    * @param value value to consider
+    * @return count of the value
+    */
+  def count(value: T): Long = getCount(sketch, value)
+
+  /**
+    * Gets the count for a given value, which has already been converted into a string. Useful
+    * if you know the string key space ahead of time.
+    *
+    * @param value value to consider, converted into an appropriate string key
+    * @return count of the value
+    */
+  def countDirect(value: String): Long = sketch.estimateCount(value)
+
+  /**
+    * Gets the count for a given value, which has already been converted into a long. Useful
+    * if you know the long key space ahead of time (e.g. with z-values).
+    *
+    * @param value value to consider, converted into an appropriate long key
+    * @return count of the value
+    */
+  def countDirect(value: Long): Long = sketch.estimateCount(value)
+
+  /**
+    * Number of observations in the frequency map
+    *
+    * @return number of observations
+    */
+  def size: Long = sketch.size()
+
+  override def observe(sf: SimpleFeature): Unit = {
+    val value = sf.getAttribute(attribute)
+    if (value != null) {
+      addAttribute(sketch, value.asInstanceOf[T])
+    }
+  }
+
+  // no-op
+  override def unobserve(sf: SimpleFeature): Unit = {}
+
+  override def +(other: Frequency[T]): Frequency[T] = {
+    val plus = new Frequency[T](attribute, precision, sketch.getRelativeError, sketch.getConfidence)
+    plus += this
+    plus += other
+    plus
+  }
+
+  override def +=(other: Frequency[T]): Unit = new RichCountMinSketch(sketch).add(other.sketch)
+
+  override def clear(): Unit = new RichCountMinSketch(sketch).clear()
+
+  override def isEmpty: Boolean = sketch.size == 0
+
+  override def toJson: String = {
```

Comment:
Ideally it would show the whole stat - when calling through WPS for example, you generally get back the json. But for stats like this one it is hard to represent as a string...